inc/environment_indicator/namespace.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+namespace Altis\Cloud\Environment_Indicator;
+
+use function Altis\get_environment_name;
+use function Altis\get_environment_type;
+
+function bootstrap() {
+	add_action( 'wp_before_admin_bar_render', __NAMESPACE__ . '\\add_admin_bar_env_info' );
+}
+
+/**
+ * Add menu item to admin bar to display environment info:
+ * - environment type
+ * - environment name
+ * - URL to Altis dashboard for the environment
+ */
+function add_admin_bar_env_info() {
+	if ( ! is_admin_bar_showing() ) {
+		return;
+	}
+
+	// Specify capabilities for which environment indicator is shown in the admin bar.
+	$capabilities[] = 'manage_options';
+
+	if ( is_multisite() ) {
+		$capabilities[] = 'manage_network_options';
+	}
+
+	$capabilities = apply_filters( 'altis_env_indicator_capabilities', $capabilities );
+	foreach ( $capabilities as $cap ) {
+		if ( ! current_user_can( $cap ) ) {
+			return;
```

Comment:
Is the intention that the user should have all of the capabilities listed in `altis_env_indicator_capabilities` to be able to see it? This will stop executing the function as soon as a user lacks even one of the specified capabilities which I'm not sure is what you're going for.

I think it might be better to approach this the other way around. Check for a capability called `view_environment_indicator` or `is_super_admin()`.

Reply:
Hm, first of all I couldn't find the `view_environment_indicator` capability in the code. Where is it defined?
I would argue that we should display it for all `admins` not just `super_admins` as `admins` still can make changes with side effects they better be sure and notified what env they are on. So, should we keep it to `manage_options`?

Reply:
A capability is just a string passed to `current_user_can()`, by doing that you've essentially created a capability that developer can assign to different roles. There's nothing more to do.

Good point re. single site admins. To be honest I'm not totally sure - a super admin is more likely to have access to staging and development sites and will have the context for what it means to be on production, whereas a single site admin is more likely to be working on production by default anyway. The goal is really to hide this from people who are most likely to just be doing their day to day work.

My thinking re. super admins is because all Altis sites are multisite by default. How about we default to super admins and make it a configuration option to show the indicator regular site admins too.